language/move-vm/runtime/src/interpreter.rs

```diff
@@ -54,53 +53,47 @@ macro_rules! debug_writeln {
 ///
 /// An `Interpreter` instance is a stand alone execution context for a function.
 /// It mimics execution on a single thread, with an call stack and an operand stack.
-/// The `Interpreter` receives a reference to a data store used by certain opcodes
-/// to do operations on data on chain and a `TransactionMetadata` which is also used to resolve
-/// specific opcodes.
-pub(crate) struct Interpreter<'txn> {
+pub(crate) struct Interpreter {
     /// Operand stack, where Move `Value`s are stored for stack operations.
     operand_stack: Stack,
     /// The stack of active functions.
     call_stack: CallStack,
-    /// Transaction data to resolve special bytecodes (e.g. GetTxnSequenceNumber, GetTxnPublicKey,
-    /// GetTxnSenderAddress, ...)
-    txn_data: &'txn TransactionMetadata,
+    /// The sender of the transaction. This can materialize as an argument to a script.
+    sender: AccountAddress,
 }
 
-impl<'txn> Interpreter<'txn> {
+impl Interpreter {
     /// Entrypoint into the interpreter. All external calls need to be routed through this
     /// function.
     pub(crate) fn entrypoint(
-        data_store: &mut dyn DataStore,
-        loader: &Loader,
-        txn_data: &'txn TransactionMetadata,
-        cost_strategy: &mut CostStrategy,
         function: Arc<Function>,
         ty_args: Vec<Type>,
         args: Vec<Value>,
+        sender: AccountAddress,
+        txn_size: AbstractMemorySize<GasCarrier>,
```

Comment:
Do we want to have the charging logic on caller side or callee side? IMHO we can do it over the caller side as we really need to charge the gas only when running the script body. But it's fine if we decide to do it in a separate PR.

Reply:
the idea is to hav a separate PR where we do that only and see the alternative.
I think there are a couple of options and I think @tzakian may like one and @runtian-zhou (and possibly me ;)) another.
PR (or multiple ones) coming soon